Support for ext. Wpf Toolkit Controls

jogibear9988 opened the pull request:
fixes #30

jogibear9988 commented:
![image](https://cloud.githubusercontent.com/assets/364896/5636184/1ef20faa-95f0-11e4-9644-b8f3dc2d38f9.png)

Maybe you canlook after merge:

I think the Controls should stretch, but I'm not sure what look and feel Standart should be!

And should the Cells be Editable only when Editmode is active?

Look and change how you like!

objorke commented on Source/PropertyTools.Wpf/DataGrid/ControlFactory/DataGridControlFactory.cs:
```diff
@@ -55,6 +70,23 @@ public virtual FrameworkElement CreateDisplayControl(PropertyDefinition property
         /// </returns>
         public virtual FrameworkElement CreateEditControl(PropertyDefinition propertyDefinition, string bindingPath)
         {
+            if (propertyDefinition is TemplateColumnDefinition)
```
I think it is time to refactor so the `CreateEditControl` method is a virtual method on the `PropertyDefinition`. Make it open for extension!

objorke commented on Source/PropertyTools.Wpf/DataGrid/ControlFactory/DataGridControlFactory.cs:
```diff
@@ -31,6 +31,21 @@ public class DataGridControlFactory : IDataGridControlFactory
         /// </returns>
         public virtual FrameworkElement CreateDisplayControl(PropertyDefinition propertyDefinition, string bindingPath)
         {
+            if (propertyDefinition is TemplateColumnDefinition)
```
Similar to line 73

objorke commented:
It would also be great to have it included in the appveyor solution!

objorke commented:
And a new NuGet package. Simply create the .nuspec and update `appveyor.yml`

jogibear9988 commented:
I've Updated.
Also I fixed the AppVeyor Solution!

For the Nuspec. To fix it, I first need to create a Nuget for the modified Extened WPF Toolkit which I use (https://github.com/dotnetprojects/WpfExToolkit)
I use this one (it's from me), because the original one has some Bug's not fixed, and this one also has a few Additional Controls (BrushEditor, TokenizedTextBox,...)

objorke commented on Source/PropertyTools.Wpf.ExtendedToolkit/PropertyTools.Wpf.ExtendedToolkit_NET45.csproj:
```diff
@@ -0,0 +1,69 @@
+    <Reference Include="System.Xml" />
+    <Reference Include="WindowsBase" />
+    <Reference Include="Xceed.Wpf.Toolkit">
+      <HintPath>libs\Xceed.Wpf.Toolkit.dll</HintPath>
```
don't add dlls under the `~/Source` folder

objorke replied:
alt 1: reference a NuGet package
alt 2: move to `~/Dependencies/Xceed.Wpf.Toolkit/Xceed.Wpf.Toolkit.dll`

objorke replied:
I understand this is not the offical `Xceed.Wpf.Toolkit`, then this dependency should have a different name. Otherwise I guess it will not be possible to use this in an application where the official toolkit is used?

jogibear9988 commented:
I've not had had time to fix this yet, but it's not forgotten!

objorke commented:
This PR is outdated - please reopen this PR or create a new PR when the feature is ready! :+1: :smiley:

jogibear9988 commented:
I've created a Nuget yesterday! Can you open the Pull Req again, I can't do it!

jogibear9988 commented:
I create a new One!!!
